paasta_tools/generate_deployments_for_service.py

```diff
@@ -184,6 +184,8 @@ def get_desired_state(service, branch, remote_refs, deploy_group):
     # paasta-paasta-cluster.instance
     tag_pattern = r'^refs/tags/(?:paasta-){1,2}%s-(?P<force_bounce>[^-]+)-(?P<state>.*)$' % branch
 
+    valid_states = ['start', 'stop']
```

Comment:
I think this is more efficient if you put this in the regex in line 185 `(start|stop)`

That way you are only even looking at refs you care about